articles/extending-classes/index.md

```diff
@@ -0,0 +1,189 @@
+- This also leads to faster progression and development time.
+- Code enhances maintenance and memory utilization.
+
+### Visibility Modes
+**Visibility mode defines whether the properties of super class are privately derived, publicly derived or protected derived.** The visibility mode plays an important role in controlling the access specifiers. They make sure that the properties are inherited accurately to the derived class. The properties of the base class are inherited in the following manner:
+
+![Visibility Modes](/engineering-education/extending-classes/ec.jpg)
+
+The above chart shows how properties are inherited and in which state they will be if the base class and derived class are private, public or protected respectively.
+
+### Defining Derived Classes
+Declaring a subclass or derived class is as easy as declaring a superclass. By using a colon(:) after class(subclass) name, one can indicate that it(subclass) has been derived from another class(superclass).
+
+##### The Syntax:
+
+>class subclass_name:visibility_mode superclass_name
+{
+	    *...members of the derived class...*
+};
+
+This is the syntax to declare a derived class. Here,
+**class** is the keyword.
+**subclass_name** is the name of the derived class.
+**:** is the symbol that indicates that the subclass is derived from superclass.
+**visibility_mode** defines the kind of derivation.
+**superclass_name** is the name of the base class.
+
+Like the superclass, the subclass holds its own data members and member functions(mentioned as *...members of the derived class…* in the syntax). Also the declaration of the subclass should be ended by using a **semicolon(;)**.
+**Note**: If visibility mode is not mentioned, then it is considered to be private by default.
+
+### Examples for derived classes:
+Consider **abc()** and **def()** as base class and derived class with two data members of **int** and **char** type respectively. The two member functions for base class will be **demo1()** and **demo2()** and two member functions for derived class will be **demo3()** and **demo4()**.
+
+- ##### Private Derived Class
+```
+class abc
+{
+				private:
+					int x;
+					char y[10];
+				public:
+					void demo1();
+					void demo1();
+};
+class def : private abc
+{
+		private:
+			int p;
+			char q[10];
+		public:
+			void demo3();
+			void demo4();
+};
+```
+
+- ##### Public Derived Class
+```
+class abc
+{
+				private:
+					int x;
+					char y[10];
+				public:
+					void demo1();
+					void demo1();
+};
+class def : public abc
+{
+		private:
+			int p;
+			char q[10];
+		public:
+			void demo3();
+			void demo4();
+};
+```
+
+- ##### Protected Derived Class
+```
+class abc
+{
+				private:
+					int x;
+					char y[10];
+				public:
+					void demo1();
+					void demo1();
+};
+class def : protected abc
+{
+		private:
+			int p;
+			char q[10];
+		public:
+			void demo3();
+			void demo4();
+};
+```
+
+### Differentiating between Derived Classes
```

Comment:
@kvardhan10 The below 3 paragraphs all appear to be the same. Can you please sufficiently differentiate these from one another?